include/photos.php

```diff
@@ -42,16 +45,28 @@ function photo_albums($uid, $update = false) {
 	$key = "photo_albums:".$uid.":".local_user().":".remote_user();
 	$albums = Cache::get($key);
 	if (is_null($albums) OR $update) {
-		/// @todo This query needs to be renewed. It is really slow
-		// At this time we just store the data in the cache
-		$albums = qu("SELECT COUNT(DISTINCT `resource-id`) AS `total`, `album`
-			FROM `photo` USE INDEX (`uid_album_created`)
-			WHERE `uid` = %d  AND `album` != '%s' AND `album` != '%s' $sql_extra
-			GROUP BY `album` ORDER BY `created` DESC",
-			intval($uid),
-			dbesc('Contact Photos'),
-			dbesc(t('Contact Photos'))
-		);
+		if (!Config::get('system', 'no_count', false)) {
+			/// @todo This query needs to be renewed. It is really slow
```

Comment:
Our huge problem is that this whole stuff is so dynamic because of the permissions. We shouldn't work on the caching but on the query. I haven't worked with views by now. But maybe this will help on many places.